### PR TITLE
AGENT-854: day2 add-nodes workflow validations

### PR DESCRIPTION
**Pull request by andfasano**
This patch introduces a set of validations required when running the `ClusterInfo` asset as part of the day2 add-nodes workflow

### CODE REVIEW

File: pkg/asset/agent/installconfig.go [hunk 1]
```diff
@@ -78,6 +78,9 @@ func (a *OptionalInstallConfig) validateInstallConfig(ctx context.Context, insta
 	if err := a.validateSupportedPlatforms(installConfig); err != nil {
 		allErrs = append(allErrs, err...)
 	}
+	if err := a.validatePlatformsByName(installConfig); err != nil {
```
Comment (pawanpinjarkar) [inline pkg/asset/agent/installconfig.go:81]: This semms unnecessary as `a.validateSupportedPlatforms` internally calls the same function.

Comment (andfasano) [reply pkg/asset/agent/installconfig.go:81]: Thanks for spotting it! Looks like a typo

**Review by rwsu — COMMENTED** [review]
/approve

**Review by pawanpinjarkar — COMMENTED** [review]
/lgtm

**Pull request merged**